src/visual.ts

```diff
@@ -355,7 +355,7 @@ module powerbi.extensibility.visual {
             histogramLayout = d3.layout.histogram();
 
             if (settings.general.bins && settings.general.bins > HistogramGeneralSettings.MinNumberOfBins) {
-                histogramLayout = histogramLayout.bins(settings.general.bins);
+                histogramLayout = histogramLayout.bins(values.length < settings.general.bins ? values.length : settings.general.bins);
```

Comment:
What about the case when I want to show 10 columns but the data-set has 5 values. Why do we reject such cases?